### DISCUSSION TITLE
OpenMP support

### DISCUSSION THREAD

**Discussion by skuntz** [OpenCilk/opencilk-project#216] [Q&A] [answered]
I see the OpenMP directory in the OpenCilk project. Is OpenMP supported or just being carried along?

**Comment by neboat** [comment, marked as answer]
The OpenMP directory is a copy of the OpenMP directory from LLVM's repository, which the opencilk-project repository is forked from.

One can build OpenCilk project from source with OpenMP enabled and then build and run OpenMP programs with that compiler.  In that sense, OpenCilk supports OpenMP, just like how ordinary LLVM and Clang support OpenMP.  But we have not implemented special support for OpenMP programs beyond what ordinary Clang/LLVM provide.  For example, we have not modified OpenCilk's Clang front end to compile OpenMP pragmas to the Tapir IR for task parallelism.

**Reply by skuntz** [reply]
Thank you for the clarification. Can you mix Cilk with OpenMP?

**Reply by neboat** [reply]
You can write, compile, and run a program with both Cilk and OpenMP in it.  But I wouldn't recommend doing that.  The performance of that code might not be ideal, as the Cilk and OpenMP runtime systems can end up competing for parallel processing resources and slowing each other down.  There may also be ways of mixing Cilk and OpenMP constructs that result in bad program behaviors.  In general, we recommend sticking with just one parallel platform, in order to avoid multiple runtime systems getting in each other's way.
